scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -0,0 +1,465 @@
+if (!dir.exists(analysis_dir)){
+  dir.create(analysis_dir, recursive = TRUE)
+}
+
+cellinfo_file <- file.path(analysis_dir, "PBMC_TotalSeqB_cellinfo.tsv")
+```
+
+
+Read in the SingleCellExperiment
+```{r}
+sce <- readRDS(sce_file)
+```
+
+
+Let's explore what is in the SCE already:
+
+```{r}
+sce
+```
+
+Notice the `altExpNames`?
+Those are the ADT data, stored as an _Alternative Experiment_.
+To access these, we can use the `altExp()` function. 
+Let's look at what we have in that slot:
+
+```{r}
+altExp(sce, "ADT")
+```
+
+
+Ah! Another SingleCellExperiment! Inception!
+Since there is only one `altExp`, the second argument is optional, so we won't use it for the rest of this notebook.
+Here the columns are again the cells, but now the rows are the proteins that were assayed with the antibodies and tags.
+
+```{r}
+# What proteins were assayed?
+adt_tags <- rownames(altExp(sce))
+adt_tags
+```
+
+
+### Clustering redux
+
+These data don't have clusters assigned yet, so let's add those, using the PCA matrix we have for the gene expression data.
+
+
+```{r}
+# perform clustering
+nn_k <- 20
+nn_clusters <- bluster::clusterRows(
+  reducedDim(sce, "PCA"), 
+  bluster::NNGraphParam(
+    k = nn_k, 
+    type = "jaccard",
+    cluster.fun = "louvain"
+  )
+)
+
+# add clusters to colData
+sce$nn_cluster <- nn_clusters
+```
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "nn_cluster")
+```
+But what are these clusters? Do they correspond to cell types?
+
+## Cell surface markers
+
+The markers we have were presumably chosen for the fact that they mark particular cell types, so let's look at a few.
+
+CD3 is a marker of T-cells 
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD3")
+```
+
+CD4+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD4")
+```
+CD8+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD8")
+```
+
+Looking at all of the markers here, but we probably don't want to do this, as we haven't introduced `purrr` yet.
+```{r}
+all_plots <- adt_tags %>%
+  purrr::map(~ scater::plotUMAP(sce, colour_by = .))
+
+patchwork::wrap_plots(all_plots)
+```
+
+
+So one thing we could do is construct a set of rules based on ADT tags, similar to how you might set gating rules for cell sorting.
+
+
+
+What thresholds should we use?
+If this is a useful marker, we should see a bimodal plot of the marked density.
+
+We'll do a bit of wrangling to pull out the ADT data, transpose it so each row is a cell,
+then convert it into a data frame for plotting. 
+(We have to first convert to a "regular" R matrix or R won't know how to convert it to a data frame)
+
+```{r}
+# convert to a data frame
+adt_df <- logcounts(altExp(sce)) %>%
+  t() %>% 
+  as.matrix() %>%
+  as.data.frame()
+```
+Now we can make a density plot with `ggplot2`!
+
+```{r}
+ggplot(adt_df, aes(x = CD3)) + 
+  geom_density() +
+  theme_bw()
+```
+
+Looking at a few of these, we can define some rough cutoffs.
```

Comment:
Are there meant to be a few of these, specifically the CD4 and CD8 distributions for the magic cutoffs below?
 This is more easily be wrangled than the previous plot grid with `dplyr::filter()` and then faceting.

Reply:
Probably should be... I made them, but didn't save them. I didn't really want to get into faceting (I would use `pivot_longer`, probably), so I was a bit torn... But it is probably worth doing.

Reply:
Just noting that I agree with this comment and think we should show CD3, CD4, and CD8 if we are going to pick cutoffs for all of them in the chunk below.